gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Channelizer.java

```diff
@@ -65,34 +76,43 @@
     public void close(final Channel channel);
 
     /**
-     * Create a message for the driver to use as a "keep-alive" for the connection. This method will only be used if
+     * Create a message for the driver to use as a "keep-alive" for the connectionPool. This method will only be used if
      * {@link #supportsKeepAlive()} is {@code true}.
+     *
+     * This method will be deprecated in future. Keep-alive is implemented using Netty ping-pong mechanism.
+     * See {@link WebSocketIdleEventHandler}
      */
+    @Deprecated
```

Comment:
I seem to remember that the point of this method was to provide a way to create keep-alive messages specific to the `Channelizer` implementation. If websockets no longer needs that because of `WebSocketIdlEventHandler` then that's fine, but I'm not sure that there will never be a need for this. On the other hand, I wonder if we'll realistically ever use this abstraction anyway - this driver pretty much just does websockets and that's it. Maybe all such abstraction could be deprecated...not sure. Thoughts?

Reply:
I am in favor of coupling the driver's implementation with WebSocket along with providing hooks for providers to inject additional logic (Right now, they do it by overriding the `Channelizer` by a custom implementation which extends the `AbstractChannelizer`). This would simplify code and provide flexibility.

We can extract out an interface (if required) in future releases when we have a concrete implementation of an alternative protocol (HTTP/2 maybe?)

Having said that, I am completely fine with not deprecating this method and keeping it as it is, if you think that is the way to go.

Reply:
after thinking on this for a bit, let's not deprecate this method until we specifically want to close down the entire abstraction.

Reply:
ok